src/twisted/mail/maildir.py

```diff
@@ -261,12 +264,13 @@ def startMessage(self, user):
             name, domain = user.split('@', 1)
         else:
             name, domain = user.dest.local, user.dest.domain
-        dir = self.userDirectory(name)
+        dirPath = FilePath(self.userDirectory(name))
         fname = _generateMaildirName()
-        filename = os.path.join(dir, 'tmp', fname)
-        fp = open(filename, 'w')
-        return MaildirMessage('%s@%s' % (name, domain), fp, filename,
-                              os.path.join(dir, 'new', fname))
+        filename = dirPath.child('tmp').child(fname).asBytesMode().path
+        fp = open(filename, 'wb')
```

Comment:
Same comment as with `initializeMaildir()`: I think if we're converting to `FilePath`s, it's appropriate to use the FilePath methods instead of the `os` module.

In any event, the `asBytesMode()` seems suspect to me.